Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -625,6 +629,27 @@ public class RestRequest {
 
         return nil
     }
+
+    /// Method to identify the charset encoding defined by the Content-Type header
+    /// - Defaults set to .utf8
+    /// - Parameter contentType: The content-type header string
+    /// - Returns: returns the defined or default String.Encoding.Type
+    private func getCharacterEncoding(from contentType: String? = nil) -> String.Encoding {
+        guard let text = contentType,
+              let regex = try? NSRegularExpression(pattern: "(?<=charset=).*(?=$|;)"),
```

Comment:
This regex is not a complete solution, but I think it will catch most valid inputs. I would suggest tweaking the `.*` to `.*?` and then post-processing the matched string to trim any surrounding whitespace and `"` marks if they exist.
Relevant RFC: https://tools.ietf.org/html/rfc2045#section-5.1